File: Problems/No.1343_Number_of_Sub_arrays_of_Size_K_and_Average_Greater_than_or_Equal_to_Threshold/number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold.py
class Solution(object):
    def numOfSubarrays(self, arr, k, threshold):
        """
        :type arr: List[int]
        :type k: int
        :type threshold: int
        :rtype: int
        """

        # parameters
        target_threshold = k*threshold
        k_sum, cleared_subarrays = 0, 0

        # start computing
        for i in range(len(arr)):
            k_sum += arr[i]
            if (i < k-1):
                continue
            else:
                if (k_sum >= target_threshold):
                    cleared_subarrays += 1
                k_sum -= arr[i-k+1]
        
        return cleared_subarrays

File: Problems/No.1343_Number_of_Sub_arrays_of_Size_K_and_Average_Greater_than_or_Equal_to_Threshold/number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        
        // parameter
        int target_threshold = k*threshold;
        int k_sum = 0;
        int cleared_subarrays = 0;
        
        // start computing
        for(int i=0; i<arr.size(); i++) {

            k_sum += arr[i];
            if (i < k-1) continue;
            else {
                
                if (k_sum >= target_threshold) cleared_subarrays++;
                k_sum -= arr[i-k+1];

            }

        }

        return cleared_subarrays;

    }
};
